src/Storages/RocksDB/StorageEmbeddedRocksDB.cpp

```diff
@@ -448,15 +452,28 @@ void StorageEmbeddedRocksDB::initDB()
         }
     }
 
-    status = rocksdb::DB::Open(merged, rocksdb_dir, &db);
-
-    if (!status.ok())
+    if (ttl > 0)
     {
-        throw Exception(ErrorCodes::ROCKSDB_ERROR, "Fail to open rocksdb path at: {}: {}",
-            rocksdb_dir, status.ToString());
+        rocksdb::DBWithTTL * db;
+        status = rocksdb::DBWithTTL::Open(merged, rocksdb_dir, &db, ttl, read_only);
+        if (!status.ok())
+        {
+            throw Exception(ErrorCodes::ROCKSDB_ERROR, "Fail to open rocksdb path at: {}: {}",
+                rocksdb_dir, status.ToString());
+        }
+        rocksdb_ptr = std::unique_ptr<rocksdb::DBWithTTL>(db);
+    }
+    else
+    {
+        rocksdb::DB * db;
+        status = rocksdb::DB::Open(merged, rocksdb_dir, &db);
```

Comment:
```suggestion
        if (read_only)
            status = rocksdb::DB::OpenForReadOnly(merged, rocksdb_dir, &db);
        else
            status = rocksdb::DB::Open(merged, rocksdb_dir, &db);
```

Reply:
can we need to support Engine=EmbeddedRocksDB(true)?